crates/gpui/src/geometry.rs

```diff
@@ -2831,9 +2837,76 @@ impl From<usize> for Pixels {
     Deserialize,
 )]
 #[repr(transparent)]
-pub struct DevicePixels(pub i32);
+pub struct PhysicalPixels<T>(pub T);
```

Comment:
Given that this type has some very specific relationships to other types here, do you think it's possible to make this field not `pub`?

Reply:
It's certainly possible. Although I'm not sure if it'll be meaningful -- we need to provide ways to both construct this from the scalar (i.e. `physical_px(_)`) and to get the scalar back (currently `.0`), which essentially gives you full access to the field.

Reply:
I guess one thing that making the field private might help with is searching for the accesses, it's generally esier to search for a method call than a field access.

Reply:
I also like it because it allows us to communicate the issue with using that API. Calling it `raw()` or some such lets people know they need to be on their toes.